sdks/java/core/src/test/java/org/apache/beam/sdk/io/CompressedSourceTest.java

```diff
@@ -907,30 +1052,33 @@ public void close() throws IOException {
   }
 
   /** Writes a single output file. */
-  private void writeFile(File file, byte[] input, CompressionMode mode) throws IOException {
-    try (OutputStream os = getOutputStreamForMode(mode, new FileOutputStream(file))) {
-      os.write(input);
+  private void writeFile(File file, byte[] input, Compression compression) throws IOException {
+    if (compression == Compression.SNAPPY) {
+      try (OutputStream os =
+          getOutputStreamForModeWithDecompressedSizeInfo(
```

Comment:
Please add the `byte[] input` as a new parameter of `getOutputStreamForMode` so we can better deal with the case specifics there.

Reply:
Of course. Thought of this, but I didn't want to mess with the old method signature. Glad you've suggested this. Sorted!